LRQA-77402 Add test to cover rollback of files in conjunction with the fileEntries within a transaction

kyle-miho opened the pull request.

liferay-continuous-integration commented:
CI is automatically triggering the following test suites:
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**relevant**
- &nbsp;&nbsp;&nbsp;&nbsp;ci:test:**sf**

kyle-miho commented:
@rafaprax I fixed and added test cases for rolling back failed transactions for update and deletion as well, since Dave also mentioned those possibilities. I'm most unsure with the file content update rollback, regarding optimal logic and SF if you can look at that one more. (It seems like the test correctly tests the functionality, but maybe not in a clean or optimal way)

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>sf</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-37.liferay.com/job/test-portal-source-format/3574/">test-portal-source-format</a></p><p><b>For more details click <a href="https://test-1-37.liferay.com/userContent/jobs/test-portal-source-format/builds/3574/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:x: ci:test:sf - 0 out of 1 jobs passed in 4 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/d5dbe3e5e51aa86f8c5cba36ef367fb570441425">d5dbe3e5e51aa86f8c5cba36ef367fb570441425</a></p><h4>Sender Branch:</h4><p>Branch Name: <a href="https://github.com/kyle-miho/liferay-portal/tree/master-qa-77402">master-qa-77402</a><br/>Branch GIT ID: <a href="https://github.com/kyle-miho/liferay-portal/commit/04d4e439b9d34fce59597ce8a1a859d501bbcd6a">04d4e439b9d34fce59597ce8a1a859d501bbcd6a</a></p>0 out of 1jobs PASSED<div><h4>1 Failed Jobs:</h4><ul><li><a href="https://test-1-37.liferay.com/job/test-portal-source-format/3574/"><strike>test-portal-source-format</strike></a></li></ul></div><h5>For more details click <a href="https://test-1-37.liferay.com/userContent/jobs/test-portal-source-format/builds/3574/jenkins-report.html">here</a>.</h5><pre>     [java] java.lang.Exception: Found 4 formatting issues:
     [java] 1: Name of variable with type 'InputStream' should end with 'inputStream': ./modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/service/test/DLFileEntryTransactionTest.java 176 (Checkstyle:VariableNameCheck)
     [java] 2: Name of variable with type 'InputStream' should end with 'inputStream': ./modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/service/test/DLFileEntryTransactionTest.java 197 (Checkstyle:VariableNameCheck)
     [java] 3: Name of variable with type 'InputStream' should end with 'inputStream': ./modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/service/test/DLFileEntryTransactionTest.java 209 (Checkstyle:VariableNameCheck)
     [java] 4: ./modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/service/test/DLFileEntryTransactionTest.java expected:&lt;...tStream fileStream =[ _fileSystemStore.getFileAsStream(
     [java]         TestPropsValues.getCompanyId(), _group.getGroupId(),
     [java]         _dlFileEntry.getName(), null);
     [java] 
     [java]       Assert.assertArrayEquals(
     [java]         StreamUtil.toByteArray(fileStream), originalBytes);
     [java] 
     [java]       TransactionInvokerUtil.invoke(
     [java]         _transactionConfig,
     [java]         () -&gt; \{
     [java]           _dlFileEntryLocalService.updateFileEntry(
     [java]             _dlFileEntry.getUserId(), _dlFileEntry.getFileEntryId(),
     [java]             _dlFileEntry.getFileName(), _dlFileEntry.getMimeType(),
     [java]             _dlFileEntry.getTitle(), null,
     [java]             _dlFileEntry.getDescription(), StringPool.BLANK,
     [java]             DLVersionNumberIncrease.fromMajorVersion(false),
     [java]             _dlFileEntry.getFileEntryTypeId(), new HashMap&lt;&gt;(),
     [java]             null, new ByteArrayInputStream(updatedBytes), 0, null,
     [java]             null, ServiceContextTestUtil.getServiceContext());
     [java] 
     [java]           InputStream updatedFileStream =
     [java]             _fileSystemStore.getFileAsStream(
     [java]               TestPropsValues.getCompanyId(), _group.getGroupId(),
     [java]               _dlFileEntry.getName(), null);
     [java] 
     [java]           Assert.assertArrayEquals(
     [java]             StreamUtil.toByteArray(updatedFileStream),
     [java]             updatedBytes);
     [java] </pre></details></html>

liferay-continuous-integration commented:
GitHub error message: state cannot be changed. The master-qa-77402 branch was force-pushed or recreated.

liferay-continuous-integration commented:
<div><div><strong>Jenkins Build:</strong><span></span><a href="https://test-1-37.liferay.com/job/test-portal-source-format/3574/">test-portal-source-format#3574</a><br/></div><div><strong>Jenkins Report:</strong><span></span><a href="https://test-1-37.liferay.com/userContent/jobs/test-portal-source-format/builds/3574/jenkins-report.html">jenkins-report.html</a><br/></div><div><strong>Jenkins Suite:</strong><span></span>sf<br/></div><div><strong>Pull Request:</strong><span></span><a href="https://github.com/liferay-core-infra/liferay-portal/pull/2207">liferay-core-infra#2207</a><br/></div><div><strong>Testray Routine:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/builds?testrayRoutineId=1306753894">EE Pull Request</a><br/></div><div><strong>Testray Build:</strong><span></span><a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2016064930">[master] ci:test:sf - kyle-miho &gt; liferay-core-infra - PR#2207 - 2022-10-25[14:08:55]</a><br/></div><div><strong>Testray Importer:</strong><span></span><a href="https://test-1-37.liferay.com/job/publish-testray-report/13796/">publish-testray-report#13796</a><br/></div></div>

liferay-continuous-integration commented:
<h1>Build started.</h1><p>Jenkins is currently running tests in the <code>relevant</code> test suite.</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a></p><h4>Job Summary:</h4><p>Job Link: <a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest(master)/5917/">test-portal-acceptance-pullrequest(master)</a></p><p><b>For more details click <a href="https://test-1-37.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/5917/jenkins-report.html">here</a>.</b></p>

liferay-continuous-integration commented:
<html><h3>:x: ci:test:relevant - 0 out of 1 jobs passed in 2 minutes</h3><details><summary>Click here for more details.</summary><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/liferay-portal/tree/master">master</a><br/>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/d5dbe3e5e51aa86f8c5cba36ef367fb570441425">d5dbe3e5e51aa86f8c5cba36ef367fb570441425</a></p><h4>Upstream Comparison:</h4><p>Branch GIT ID: <a href="https://github.com/liferay/liferay-portal/commit/d5dbe3e5e51aa86f8c5cba36ef367fb570441425">d5dbe3e5e51aa86f8c5cba36ef367fb570441425</a><br/>Jenkins Build URL: <a href="https://testray.liferay.com/home/-/testray/runs?testrayBuildId=2015982868">EE Development Acceptance (master) - 3945 - 2022-10-25[10:27:21]</a></p><details><summary><strong>ci:test:relevant - 0 out of 1 jobs PASSED</strong></summary><div><h4>1 Failed Jobs:</h4><ul><li><a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest(master)/5917/"><strike>test-portal-acceptance-pullrequest(master)</strike></a></li></ul></div></details><h5>For more details click <a href="https://test-1-37.liferay.com/userContent/jobs/test-portal-acceptance-pullrequest(master)/builds/5917/jenkins-report.html">here</a>.</h5><hr/><h4>Failures unique to this pull:</h4><ol><li><div><h5><a href="https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest(master)/5917/">test-portal-acceptance-pullrequest(master)</a></h5><div><h6>Job Results:</h6><p>0 Jobs Passed.<br/>1 Job Failed.</p></div><div><p>The sender branch SHA could not be found on <strong><a href="https://github.com/liferay-core-infra/liferay-portal/pull/2207">kyle-miho/master-qa-77402</a></strong>. The sender branch may have been force pushed or deleted after the pull request test was initiated.</p><pre>fatal: Not a valid branch point: '04d4e439b9d34fce59597ce8a1a859d501bbcd6a'.</pre></div></div></li></ol><h4>For upstream results, click <a href="https://test-1-1.liferay.com/job/test-portal-acceptance-upstream-dxp(master)">here</a>.</h4></details></html><h5>Test bundle downloads:</h5><ul><li><a href="https://files.liferay.com/private/bundles/test-1-37/jobs/test-portal-acceptance-pullrequest(master)/builds/5917/liferay-portal-bundle-tomcat.tar.gz">liferay-portal-bundle-tomcat.tar.gz</a> (mirrors: <a href="http://mirrors.dlc.liferay.com/files.liferay.com/private/bundles/test-1-37/jobs/test-portal-acceptance-pullrequest(master)/builds/5917/liferay-portal-bundle-tomcat.tar.gz">dlc</a>, <a href="http://mirrors.lax.liferay.com/files.liferay.com/private/bundles/test-1-37/jobs/test-portal-acceptance-pullrequest(master)/builds/5917/liferay-portal-bundle-tomcat.tar.gz">lax</a>)</li></ul>
